### DISCUSSION TITLE
Docker Pull via Dockge

### DISCUSSION THREAD

**Discussion by mrmds** [louislam/dockge#661] [Ask for Help]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

Docker pull through Dockge?

Hi I'm new to homelabs and recently loaded dockge which is very very useful.

My question is whether a docker pull request should also be made through Dockge as I've been unable to convert the pull request to a compose. Either I am doing is from the wrong area or totally missing the point. 

For reference it's the Docker GenAI stack I'm trying to pull from the Docker repository. 

Similarly I tried to pull code down from GitHub to create compose but I'm similarly getting nowhere. 

Pointing me in the right direction or to a guide I can read would be greatly appreciated. 

Thanks!

### 🐻 Dockge Version

Latest

### 💻 Operating System and Arch

Debian12

### 🌐 Browser

Chrome latest

### 🐋 Docker Version

Latest

### 🟩 NodeJS Version

Nil

**Comment by mizady** [comment]
Dockge will pull from docker hub automatically when you attempt to deploy or start the stack.  This sounds like you maybe trying to pull from a docker container registry that is not the main docker hub registry.  If that's the case then you will need to pull the image manually as dockge has no way of using custom docker registries that I am aware of.  I want to say this feature made it part way through development but I haven't seen it finished.  A feature I would also like to have.  This doesn't sound like a dockge bug, try creating the docker compose file manually and using "sudo docker compose up -d", keep in mind you need to be in the directory of the compose.yaml file when executing the command.  If you still can't, then I would look at examples of compose files for the containers you are trying to deploy.
